File: app/src/main/java/net/afterday/compas/persistency/PersistencyProvider.java
package net.afterday.compas.persistency;

import net.afterday.compas.persistency.influences.InfluencesPersistency;
import net.afterday.compas.persistency.initialState.GameStatePersistency;
import net.afterday.compas.persistency.items.ItemsPersistency;
import net.afterday.compas.persistency.player.PlayerPersistency;

/**
 * Created by Justas Spakauskas on 2/10/2018.
 */

public interface PersistencyProvider
{
    InfluencesPersistency getInfluencesPersistency();
    ItemsPersistency getItemsPersistency();
    GameStatePersistency getInitialStatePersistency();
    PlayerPersistency getPlayerPersistency();
}

File: engine/src/main/kotlin/net/afterday/compas/engine/persistency/PersistencyProvider.kt
package net.afterday.compas.engine.persistency;

import net.afterday.compas.engine.persistency.influences.InfluencesPersistency;
import net.afterday.compas.engine.persistency.initialState.GameStatePersistency;
import net.afterday.compas.engine.persistency.items.ItemsPersistency;
import net.afterday.compas.engine.persistency.player.PlayerPersistency;

interface PersistencyProvider
{
    val influencesPersistency: InfluencesPersistency
    val itemsPersistency: ItemsPersistency
    val gameStatePersistency: GameStatePersistency
    val playerPersistency: PlayerPersistency
}
